node-it/src/test/scala/com/wavesplatform/it/asset/InvokeReissueBurnAssetSuite.scala

```diff
@@ -0,0 +1,333 @@
+    }
+
+    "Non-reissuable asset could not be reissued" in {
+      val acc     = createDapp(script(simpleNonreissuableAsset))
+      val txIssue = issue(acc, method, simpleNonreissuableAsset, invocationCost(1))
+      val assetId = validateIssuedAssets(acc, txIssue, simpleNonreissuableAsset, method = method)
+      assertBadRequestAndMessage(
+        reissue(acc, method, assetId, 100500, false),
+        "Asset is not reissuable"
+      )
+    }
+  }
+
+  "Restrictions in @Callable" - {
+    val method = "@Callable"
+
+    "Issue two identical with the same nonce should not produce an error " in {
+      val acc      = createDapp(script(simpleNonreissuableAsset))
+      val txIssue1 = issue(acc, method, simpleNonreissuableAsset, invocationCost(1))
+      val txIssue2 = issue(acc, method, simpleNonreissuableAsset, invocationCost(1))
+      validateIssuedAssets(acc, txIssue1, simpleNonreissuableAsset, method = method)
+      validateIssuedAssets(acc, txIssue2, simpleNonreissuableAsset, method = method)
+    }
+
+    "Reissuing more than Long.Max should produce an error" in {
+      val acc             = createDapp(script(simpleReissuableAsset))
+      val txIssue         = issue(acc, method, simpleReissuableAsset, invocationCost(1))
+      val assetId         = validateIssuedAssets(acc, txIssue, simpleReissuableAsset, method = method)
+      val initialQuantity = simpleReissuableAsset.q
+
+      assertBadRequestAndMessage(
+        reissue(acc, method, assetId, Long.MaxValue - initialQuantity + 1, true),
+        "State check failed. Reason: Asset total value overflow"
+      )
+    }
+
+    "Burning more than current asset count should produce an error" in {
+      val acc     = createDapp(script(simpleReissuableAsset))
+      val txIssue = issue(acc, method, simpleReissuableAsset, invocationCost(1))
+      val assetId = validateIssuedAssets(acc, txIssue, simpleReissuableAsset, method = method)
+
+      assertBadRequestAndMessage(
+        burn(acc, method, assetId, simpleReissuableAsset.q + 1),
+        "State check failed. Reason: Accounts balance errors"
+      )
+    }
+
+    "Reissuing NFT asset should produce an error" in {
+      val acc     = createDapp(script(nftAsset))
+      val txIssue = issue(acc, method, nftAsset, invocationCost(1))
+      val assetId = validateIssuedAssets(acc, txIssue, nftAsset, method = method)
+
+      assertBadRequestAndMessage(
+        reissue(acc, method, assetId, 2, true),
+        "State check failed. Reason: Asset is not reissuable"
+      )
+    }
+
+    "Burn to 1, set reissuee false, reissue" in {}
+
+    "Issue 10 assets should not produce an error" in {
+      val acc = createDapp(script(simpleNonreissuableAsset))
+      val tx  = invokeScript(acc, "issue10Assets")
+
+      for (nth <- 0 to 9) validateIssuedAssets(acc, tx, simpleNonreissuableAsset, nth, "@Callable")
+    }
+
+    "Issue more than 10 assets should produce an error" in {
```

Comment:
Check the same for Reissue and Burn separately and together

Reply:
ok